Build failure on unsupported OS

I try to compile the example code in README with crate `memory-stats` on OpenBSD (current version/future 7.5) with Rust v1.73.0.

The build fails with `cargo`:
```bash
$ cargo build -v
   Compiling memory-stats v1.1.0 (/home/fox/dev/memory-stats.git)
     Running `rustc --crate-name memory_stats --edition=2021 /home/fox/dev/memory-stats.git/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=274 --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -Cdebuginfo=2 -C metadata=f5395142778a9166 -C extra-filename=-f5395142778a9166 --out-dir /home/fox/tmp/memory-stats/target/debug/deps -C incremental=/home/fox/tmp/memory-stats/target/debug/incremental -L dependency=/home/fox/tmp/memory-stats/target/debug/deps`
error[E0583]: file not found for module `platform`
  --> /home/fox/dev/memory-stats.git/src/lib.rs:52:1
   |
52 | mod platform;
   | ^^^^^^^^^^^^^
   |
   = help: to create the module `platform`, create file "/home/fox/dev/memory-stats.git/src/platform.rs" or "/home/fox/dev/memory-stats.git/src/platform/mod.rs"

error[E0428]: the name `platform` is defined multiple times
  --> /home/fox/dev/memory-stats.git/src/lib.rs:61:1
   |
52 | mod platform;
   | ------------- previous definition of the module `platform` here
...
61 | mod platform {
   | ^^^^^^^^^^^^ `platform` redefined here
   |
   = note: `platform` must be defined only once in the type namespace of this module

error[E0412]: cannot find type `MemoryStats` in this scope
  --> /home/fox/dev/memory-stats.git/src/lib.rs:62:37
   |
62 |     pub fn memory_stats() -> Option<MemoryStats> {
   |                                     ^^^^^^^^^^^ not found in this scope
   |
help: consider importing this struct
   |
62 +     use crate::MemoryStats;
   |

error[E0425]: cannot find function `memory_stats` in module `platform`
   --> /home/fox/dev/memory-stats.git/src/lib.rs:105:15
    |
105 |     platform::memory_stats()
    |               ^^^^^^^^^^^^ not found in `platform`

error[E0565]: item in `deprecated` must be a key/value pair
   --> /home/fox/dev/memory-stats.git/src/lib.rs:102:16
    |
102 |     deprecated("memory-stats doesn't support this platform!")
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Some errors have detailed explanations: E0412, E0425, E0428, E0565, E0583.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `memory-stats` (lib) due to 5 previous errors
```

After some debug, it seems that use of `cfg_attr` (lines 49-52 in `src/lib.rs`) is incorrect => redefinition of module `platform` on OS OpenBSD line 61.

See https://stackoverflow.com/questions/58150681/inline-cfg-attrfeature-x-path-file-rs-doesnt-work for explanation => using `cfg_attr`, module `platform` is always defined even on unsupported OS.

I have a local version that fixes this issue. I will submit a PR to fix it after some build checks on a supported OS (Linux Debian).

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -46,9 +46,16 @@
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
-#[cfg_attr(target_os = "windows", path = "windows.rs")]
-#[cfg_attr(any(target_os = "linux", target_os = "android"), path = "linux.rs")]
-#[cfg_attr(any(target_os = "macos", target_os = "ios"), path = "darwin.rs")]
+#[cfg(target_os = "windows")]
+#[path = "windows.rs"]
+mod platform;
+
+#[cfg(any(target_os = "linux", target_os = "android"))]
+#[path = "linux.rs"]
+mod platform;
+
+#[cfg(any(target_os = "macos", target_os = "ios"))]
+#[path = "darwin.rs"]
 mod platform;
 
 #[cfg(not(any(
@@ -59,6 +66,8 @@ mod platform;
     target_os = "ios",
 )))]
 mod platform {
+    use crate::MemoryStats;
+
     pub fn memory_stats() -> Option<MemoryStats> {
         None
     }
@@ -91,16 +100,6 @@ pub struct MemoryStats {
 /// If the current memory usage cannot be queried
 /// or `memory_stats` is run on a unsupported platform,
 /// `None` is returned.
-#[cfg_attr(
-    not(any(
-        target_os = "windows",
-        target_os = "linux",
-        target_os = "android",
-        target_os = "macos",
-        target_os = "ios",
-    )),
-    deprecated("memory-stats doesn't support this platform!")
-)]
 pub fn memory_stats() -> Option<MemoryStats> {
     platform::memory_stats()
 }

